pkg/skaffold/docker/authenticators.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/docker/cli/cli/config"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/google"
+)
+
+var authenticators = Authenticators{}
+
+// Authenticators stores an authenticator per registry.
+type Authenticators struct {
+	byRegistry map[string]*lockedAuthenticator
+	lock       sync.Mutex
+}
+
+// For retrieves the authentiator for a given image reference.
```

Comment:
This reuse is something we could reasonably do in `google.Keychain`

Reply:
Go for it! It'll make it easier for Skaffold